packages/talos_cohort_replicator/src/talos_cohort_replicator.rs

```diff
@@ -57,15 +57,18 @@ pub struct CohortReplicatorConfig {
     pub statemap_installer_threadpool: u64,
 }
 
-async fn flatten_service_result<T>(handle: JoinHandle<Result<T, ServiceError>>) -> Result<T, ServiceError> {
+async fn flatten_service_result<T>(handle: JoinHandle<Result<T, ReplicatorError>>) -> Result<T, ReplicatorError> {
     match handle.await {
         Ok(Ok(result)) => Ok(result),
         Ok(Err(err)) => Err(err),
-        Err(err) => Err(ServiceError {
-            reason: format!("handling failed with error={err:?}"),
+        Err(err) => Err(ReplicatorError {
```

Comment:
Don't know what needs to happen for this to fail. Usually if service fails to start the error is delivered via `Ok(Err)` branch.